File: data/mi1/rooms/village1.lua
roomDefinition = {
	width = 1008,
	height = 144,
	startTable = {
		--lookout = { pos = items2["village1.cliffside"].hotspot.walk_to, facing = "east"},
		scummbar = { pos = items["village1.door"].hotspot.walk_to, dir = "south"},
		village2 = { pos = items["village1.archway"].hotspot.walk_to, dir="west"},
		lookout = { pos = items["village1.cliffside"].hotspot.walk_to, dir="east"}
	},
	defaultroom = "lookout",
	depth = { type="linear_y", values= {0, 1, 144, 0} },
	scale = { type="linear_y", values= {0, 0.8, 144, 0.2}}
}

room = scumm.factory.basic_room (roomDefinition)

room:add( {
	{ pos = {0, 0,-3}, components = { { type="gfx", image="gfx/village1_1.png" }}},
	{ pos = {73, 0, 1}, components = { { type="gfx", image="gfx/village1_2.png" }}},
	{ pos = {229, 0, 1}, components = { { type="gfx", image="gfx/village1_3.png" }}},
	{ pos = {606, 0, 1}, components = { { type="gfx", image="gfx/village1_4.png" }}},
	scumm.factory.walkarea { shape = { type = "graph", 
		vertices = { {8, 71}, {23, 63}, {49, 49}, {40, 37}, {56, 24}, {84, 13}, {770, 13}, {780,34}, {1000,34} }, 
		edges = { {0,1}, {1,2}, {2,3},{3,4}, {4,5},{5,6},{6,7},{7,8} }}},
	scumm.factory.object { id="village1.archway" },
	scumm.factory.object { id="village1.cliffside" },
	scumm.factory.object { id="village1.door" },
	scumm.factory.object { id="village1.poster" },
})

local entry_cutscene = function() 
	local proom = variables._previousRoom or roomDefinition.defaultroom
	if (proom == "lookout") then
		local actions = {
			{ type = scumm.action.walkto, args = {tag="player", pos = {230, 0}}}
		}
		local s = script.make(actions)
		s.name = "_walk"
		monkey.play(s)
	end
end

table.insert(room.initstuff, glib.curry (mi.addStorekeeper, { pos = {781, 34, 0}, walkto = { items["village1.cliffside"].hotspot.walk_to }}))
table.insert(room.initstuff, entry_cutscene)

File: data/mi1_py/rooms/village1.py
import lib_py.engine as engine
import lib_py.scumm.room as room
import lib_py.scumm.actions as act
import lib_py.scumm.entity as se
import lib_py.shape as sh
from lib_py.script import Script
from lib_py.scumm.scumm import State
import example

def init():
    if engine.previous_room == 'lookout':
        sc = Script(id = '_main')
        sc.addAction (act.Walk(pos = [280,10], tag='player'))
        example.play(sc)        

def builder():
    r =  room.RoomUI(id='village1', width = 1008, height = 144)
    r.add (se.BackgroundItem(image='gfx/village1_1.png', pos = [0, 0, -1]), 'main')
    r.add (se.BackgroundItem(image='gfx/village1_2.png', pos = [73, 0, 1]), 'main')
    r.add (se.BackgroundItem(image='gfx/village1_3.png', pos = [229, 0, 1]), 'main')
    r.add (se.BackgroundItem(image='gfx/village1_4.png', pos = [606, 0, 1]), 'main')
    r.add (se.WalkArea (tag='walkarea', shape = sh.Graph(
        nodes = [ [8, 71], [23, 63], [49, 49], [40, 37], [56, 24], [84, 13], [770, 13], [780,34], [1000,34]],
        edges = [ [0, 1], [1, 2], [2, 3], [3, 4], [4, 5], [5, 6], [6, 7], [7, 8]]),
        scale = sh.LinY(y0=0, z0=0.8, y1 = 144, z1 = 0.1),
        depth = sh.LinY(y0=0,z0=1,y1=144,z1=0)), 'main')
    r.add (se.Sprite (item='village1.poster', pos = [259, 27, 0]), 'main')
    r.add (se.Sprite (item='village1.door', anim = State.variables['door.village.scummbar'], model='door_village_scummbar', pos = [699, 7, 0]), 'main')
    r.addDynamicItems()
    r.init.append(init)
    print (State.items['scummbar.door.out'].walkto)
    return r

engine.addRoom (id = 'village1', f=builder)
